pkg/controllers/options.go

```diff
@@ -0,0 +1,46 @@
+/*
+Package controllers contains logic for handlers that belong to controllers and options for configuring controllers.
+*/
+package controllers
+
+import (
+	"os"
+	"strings"
+
+	"github.com/rancher/lasso/pkg/controller"
+)
+
+type controllerContextType string
+
+const (
+	User       controllerContextType = "user"
+	Scaled     controllerContextType = "scaled"
+	Management controllerContextType = "mgmt"
+)
+
+// GetOptsFromEnv configures a SharedControllersFactoryOptions using env var and return a pointer to it.
+func GetOptsFromEnv(contextType controllerContextType) *controller.SharedControllerFactoryOptions {
+	if !SkipUpdateOnNoChange(contextType) {
+		return nil
+	}
+	return &controller.SharedControllerFactoryOptions{
+		SkipNoChangeUpdate: true,
+	}
+}
+
+// SkipUpdateOnNoChange returns whether the env var CATTLE_SKIP_NO_CHANGE_UPDATE indicates that controlelrs for the
+// given context type should skip running enqueue if the event triggering the update func is not actual update.
+func SkipUpdateOnNoChange(option controllerContextType) bool {
```

Comment:
Could this be unexported?

Reply:
Yes it can be unexported will change. Gives us a way to isolate which set of controllers are making an impact. Also gives us flexibility if a certain set of controllers ends up having issue.